### ISSUE TITLE
Build IQueuedJob

### ISSUE THREAD

**Issue by jamesmh** [jamesmh/coravel#15]
This contract will allow queuing objects instead of methods / actions.

Will allow object serialization also, a needed step for persistance, failure re-try, etc.

```c#
IQueuedJob job = new SomeQueuedItem(user); // Inject user object to be used in job....
queue.QueueJob(job);
```

**Comment by jerriep** [comment]
What would the advantage of this be over the following?

```
queue.QueueTask(() =>
{
    var item = new SomeQueuedItem(user);
    item.Whatever();
});

```

**Comment by jamesmh** [comment]
By using the example I gave, we could store the private data/fields of the class (by exposing a "serialize()" method most likely). We would store that and the class details (name, assembly/namespace). This way, we could store the queued item for later (in 30 seconds - the queue length right now), or even expose another api / contract to schedule the item for a later date in the future. When grabbing the task from the queue (which might be ina DB or some other medium) we just need to deserialize that object and invoke it.

If we were to use an anonymous method we wouldn't be able to store / retrieve the entire command / task. We could (if we wanted) store the pre-compiled code of the anonymous method (using Expression<>), but we wouldn't be able to capture any references to non-local variables etc.

Make sense? 

P.S, I moved and had no internet for a week :(
